From 59bd7d23b80a92631984a45ffe0accab426f9625 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Sat, 27 Jun 2020 12:29:39 +0800
Subject: [PATCH 5/8] i2c: imx: narrow down i2c_imx_init_recovery_info
 definition

To fix below warning:
drivers/i2c/busses/i2c-imx.c:1151:12: warning: 'i2c_imx_init_recovery_info' defined but not used [-Wunused-function]

1151 | static int i2c_imx_init_recovery_info(struct imx_i2c_struct *i2c_imx,
     |            ^~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/i2c/busses/i2c-imx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/i2c/busses/i2c-imx.c b/drivers/i2c/busses/i2c-imx.c
index 82d7c49b8571..dbfde54bf8c7 100644
--- a/drivers/i2c/busses/i2c-imx.c
+++ b/drivers/i2c/busses/i2c-imx.c
@@ -1123,6 +1123,7 @@ static int i2c_imx_xfer(struct i2c_adapter *adapter,
 	return (result < 0) ? result : num;
 }
 
+#ifndef CONFIG_ARCH_LAYERSCAPE
 static void i2c_imx_prepare_recovery(struct i2c_adapter *adap)
 {
 	struct imx_i2c_struct *i2c_imx;
@@ -1187,6 +1188,7 @@ static int i2c_imx_init_recovery_info(struct imx_i2c_struct *i2c_imx,
 
 	return 0;
 }
+#endif
 
 /*
  * switch SCL and SDA to their GPIO function and do some bitbanging
-- 
2.17.1

